core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/spark/SparkIngestIngestDriver.java

```diff
@@ -0,0 +1,717 @@
+					configProperties,
+					DataStorePluginOptions.getStoreNamespace(inputStoreName),
+					configFile)) {
+				throw new ParameterException(
+						"Cannot find store name: " + inputStoreLoader.getStoreName());
+			}
+			inputStoreOptions = inputStoreLoader.getDataStorePlugin();
+		}
+
+		// Attempt to load input store.
+		if (inputStoreOptions == null) {
+			final StoreLoader inputStoreLoader = new StoreLoader(
+					inputStoreName);
+			if (!inputStoreLoader.loadFromConfig(
+					configProperties,
+					DataStorePluginOptions.getStoreNamespace(inputStoreName),
+					null)) {
+				throw new ParameterException(
+						"Cannot find store name: " + inputStoreLoader.getStoreName());
+			}
+			inputStoreOptions = inputStoreLoader.getDataStorePlugin();
+		}
+		// Load the Indexes
+		if (indexOptions == null) {
+			final IndexLoader indexLoader = new IndexLoader(
+					indexList);
+			if (!indexLoader.loadFromConfig(configProperties)) {
+				throw new ParameterException(
+						"Cannot find index(s) by name: " + indexList);
+			}
+			indexOptions = indexLoader.getLoadedIndexes();
+		}
+
+		// first collect the local file ingest plugins
+		final Map<String, LocalFileIngestPlugin<?>> localFileIngestPlugins = new HashMap<String, LocalFileIngestPlugin<?>>();
+		final List<WritableDataAdapter<?>> adapters = new ArrayList<WritableDataAdapter<?>>();
+		for (Entry<String, LocalFileIngestPlugin<?>> pluginEntry : ingestPlugins.entrySet()) {
+
+			if (!checkIndexesAgainstProvider(
+					pluginEntry.getKey(),
+					pluginEntry.getValue(),
+					indexOptions)) {
+				continue;
+			}
+
+			localFileIngestPlugins.put(
+					pluginEntry.getKey(),
+					pluginEntry.getValue());
+
+			adapters.addAll(Arrays.asList(pluginEntry.getValue().getDataAdapters(
+					ingestOptions.getVisibility())));
+		}
+
+		DataStore dataStore = inputStoreOptions.createDataStore();
+		try (LocalIngestRunData runData = new LocalIngestRunData(
+				adapters,
+				dataStore)) {
+
+			List<PluginVisitor> pluginVisitors = new ArrayList<PluginVisitor>(
+					localFileIngestPlugins.size());
+			for (final Entry<String, LocalFileIngestPlugin<?>> localPlugin : localFileIngestPlugins.entrySet()) {
+				pluginVisitors.add(new PluginVisitor(
+						localPlugin.getValue(),
+						localPlugin.getKey(),
+						localInput.getExtensions()));
+			}
+
+			while (inputFiles.hasNext()) {
+				final URL file = inputFiles.next().toURL();
+				for (final PluginVisitor visitor : pluginVisitors) {
+					if (visitor.supportsFile(file)) {
+						processFile(
+								file,
+								visitor.typeName,
+								visitor.localPlugin,
+								runData,
+								indexOptions,
+								ingestOptions);
+					}
+				}
+			}
+
+		}
+		catch (MalformedURLException e) {
+			LOGGER.error(
+					"Error in converting input path to URL for " + inputFiles,
+					e);
+			throw new MalformedURLException(
+					"Error in converting input path to URL for " + inputFiles);
+		}
+	}
+
+	protected void processFile(
```

Comment:
any chance to reuse some code here a little more?  I understand everything has to be serializable